pkg/operation/botanist/controlplane/kubeapiserver/deployment/kube_apiserver.go

```diff
@@ -0,0 +1,761 @@
+	vpnPort = "4314"
+
+	// constants for CA bundle mounts
+	volumeMountNameCABundleFedoraRHEL6Openelec = "fedora-rhel6-openelec-cabundle"
+	volumeMountPathCABundleFedoraRHEL6Openelec = "/etc/pki/tls"
+	volumeMountNameCABundleCentOSRHEl7         = "centos-rhel7-cabundle"
+	volumeMountPathCABundleCentOSRHEl7Dir      = "/etc/pki/ca-trust/extracted/pem"
+	volumeMountNameCABundleEtcSSL              = "etc-ssl"
+	volumeMountPathCABundleEtcSSL              = "/etc/ssl"
+	volumeMountNameCABundleUsrShareCacerts     = "usr-share-cacerts"
+	volumeMountPathCABundleUsrShareCacerts     = "/usr/share/ca-certificates"
+	volumeMountNameCABundleDebianFamily        = "debian-family-cabundle"
+	volumeMountPathCABundleDebianFamily        = "/etc/ssl/certs/ca-certificates.crt"
+	volumeMountNameCABundleFedoraRHEL6         = "fedora-rhel6-cabundle"
+	volumeMountPathCABundleFedoraRHEL6         = "/etc/pki/tls/certs/ca-bundle.crt"
+	volumeMountNameCABundleOpensuse            = "opensuse-cabundle"
+	volumeMountPathCABundleOpensuse            = "/etc/ssl/ca-bundle.pem"
+	volumeMountNameCABundleOpenelec            = "openelec-cabundle"
+	volumeMountPathCABundleOpenelec            = "/etc/pki/tls/cacert.pem"
+	volumeMountPathCABundleCentOSRHEL7File     = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+	volumeMountNameCABundleAlpine              = "alpine-linux-cabundle"
+	volumeMountPathCABundleAlpine              = "/etc/ssl/cert.pem"
+)
+
+// KubeAPIServer contains functions for a kube-apiserver deployer.
+type KubeAPIServer interface {
+	component.DeployWaiter
+	component.MonitoringComponent
+	// SetSecrets sets the secrets.
+	SetSecrets(Secrets)
+	// SetHealthCheckToken sets the health check token for the kube-apiserver
+	SetHealthCheckToken(string)
+	// SetShootAPIServerClusterIP sets the Cluster IP of the service of the shoot apiserver
+	SetShootAPIServerClusterIP(string)
+	// SetShootOutOfClusterAPIServerAddress sets the internal API server domain of the Shoot API Server
+	// used by Gardener system components
+	SetShootOutOfClusterAPIServerAddress(string)
+}
+
+// New creates a new instance of DeployWaiter for the kube-apiserver.
+func New(
+	config *gardencorev1beta1.KubeAPIServerConfig,
+	managedSeedAPIServer *gardencorev1beta1helper.ShootedSeedAPIServer,
+	seedClient kubernetes.Interface,
+	gardenClient client.Client,
+	shootKubernetesVersion *semver.Version,
+	seedNamespace string,
+	gardenNamespace string,
+	hibernationEnabled bool,
+	konnectivityTunnelEnabled bool,
+	etcdEncryptionEnabled bool,
+	basicAuthenticationEnabled bool,
+	hvpaEnabled bool,
+	mountHostCADirectories bool,
+	shootHasDeletionTimestamp bool,
+	serviceNetwork *net.IPNet,
+	podNetwork *net.IPNet,
+	nodeNetwork *net.IPNet,
+	minNodeCount int32,
+	maxNodeCount int32,
+	shootAnnotations map[string]string,
+	maintenanceWindow *gardencorev1beta1.MaintenanceTimeWindow,
+	sniValues APIServerSNIValues,
+	images APIServerImages,
+) KubeAPIServer {
```

Comment:
This is somewhat "messy" and way too many arguments. Can you redesign this and have something like this:

https://github.com/gardener/gardener/blob/555ac170fb55e66fc1389c36d8c3fcc010a05882/pkg/operation/botanist/extensions/infrastructure/infrastructure.go#L58-L87